src/services/session.service.ts

```diff
@@ -0,0 +1,79 @@
+import { ActiveProxyAddresses, ProxyBindings } from "../@types/types";
+import { listParticipantConversations, addParticipant } from "../utils";
+
+export const getActiveProxyAddresses = async (phoneNumbers: Array<String>) : Promise<ActiveProxyAddresses> => {
+  let activeConversations = {}
+
+  const promises = phoneNumbers.map(async (phoneNumber: string) => {
+    const participantConversations = await listParticipantConversations(phoneNumber)
+
+    const proxyAddresses = participantConversations.map((participant) => {
+      return participant.participantMessagingBinding.proxy_address
+    })
+
+    return activeConversations[phoneNumber] = proxyAddresses
+  })
+
+  return Promise.all(promises)
+    .then(() => {
+      return activeConversations;
+    })
+    .catch(err => {
+      console.log(err)
+      throw new Error(err)
+    })
+}
+
+export const matchAvailableProxyAddresses = async (activeProxyAddresses: ActiveProxyAddresses) : Promise<ProxyBindings> => {
+  try {
+    const phoneNumbers: Array<String> = JSON.parse(process.env.NUMBER_POOL).sort()
+    
+    let proxyBindings = {};
+  
+    for (const [key, activeAddresses] of Object.entries(activeProxyAddresses)) {
+  
+      const availableNumbers = phoneNumbers.filter((pn) => {     
+        return !activeAddresses.includes(pn);
+      })
+  
+      if (availableNumbers.length < 1) {
+        throw new Error('Not enough numbers available in pool.')
```

Comment:
Figure out better way to handle not enough numbers in pool. Currently results in UnhandledPromiseRejection.